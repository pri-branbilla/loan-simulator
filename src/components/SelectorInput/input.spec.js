import React from 'react'
import { SelectorInput } from '.'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import { mount, configure } from 'enzyme'

configure({adapter: new Adapter()})

it('renders correctly', () => {
  const tree = renderer.create(
    <SelectorInput
      inputId="inputId"
      label="Test"
      onChange={() => {}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('test input text value change', () => {
  let inputValue = 'test'
  const changeValue = (name, value) => inputValue = value
  const tree = mount(
    <SelectorInput
      inputId="inputId"
      label="Test"
      onChange={changeValue}
    />
  )
  tree.find('input').at(0).simulate('change', { target: { value: 'test 234234' } })
  expect(inputValue).toBe('test 234234')

  tree.find('input').at(1).simulate('change', { target: { value: 'test 1111' } })
  expect(inputValue).toBe('test 1111')
})
