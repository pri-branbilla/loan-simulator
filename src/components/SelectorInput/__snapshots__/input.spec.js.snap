// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="field-group"
>
  <div
    className="field"
  >
    <label
      htmlFor="inputId"
    >
      Test
    </label>
    <input
      id="inputId"
      name="inputId"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <div
      className="range"
    >
      <input
        id="inputId"
        name="inputId"
        onChange={[Function]}
        step="10"
        type="range"
      />
      <div
        className="range__values"
      >
        <span>
          R$NaN
        </span>
        <span>
          R$NaN
        </span>
      </div>
    </div>
  </div>
</div>
`;
